hrldas/IO_code/module_NoahMP_hrldas_driver.F

```diff
@@ -767,8 +767,9 @@ subroutine land_driver_ini(NTIME_out,wrfits,wrfite,wrfjts,wrfjte)
 !---------------------------------------------------------------------
 
   update_lai = .true.   ! default: use LAI if present in forcing file
-  if (dynamic_veg_option == 2 .or. dynamic_veg_option == 5 .or. &
-      dynamic_veg_option == 6 .or. dynamic_veg_option == 10) &
+  if (dynamic_veg_option == 1 .or. dynamic_veg_option == 2 .or. &
+      dynamic_veg_option == 3 .or. dynamic_veg_option == 4 .or. &
+      dynamic_veg_option == 5 .or. dynamic_veg_option == 6) &    ! remove dveg=10 and add dveg=1,3,4 into the update_lai flag false condition
```

Comment:
the comment on the end of this line is not necessary, that is more appropriate for the commit message

are you sure that 1,3,4 should be added?

Reply:
Hi Mike, this is for the update_lai flag false condition. So, for dveg=1~6, update_lai should be false. For dveg=7~9, update_lai flag should be true. 
This update_lai flag should be set to false for dveg=1~6 for cropland grids. Otherwise, the model will use the initial LAI from the setup file in the ENERGY subroutine. Although LAI will be calculated in the CARBON_CROP subroutine, it will be replaced by initial value again in the next timestep.

Reply:
If opt=1,3,4 are not included in this if block, then update_lai will be .true. for these options. Then, the LAI for each time step will be the same as the last time step (done by READFORC_HRLDAS subroutine), which will not change anything if crop_opt=0, because for opt=1,3,4, the LAI will be replaced at each time step by table values in phenology subroutine. But when crop_opt>0, the LAI will not be replaced by phenology subroutine calculations, so the initial LAI in the hrldas_setup file will be used every time step for the energy and water subroutine calculations before crop carbon calculations. As a result, LAI will be temporarily changed to the value computed from carbon subroutine after energy and water calculations every time step, and then goes back to the last time step value (i.e., initial value). Overall, this will lead to the inconsistent issue Zhe brought up.

Reply:
@CharlesZheZhang I agree with Mike that you can remove the comment at the end of this line.